lib/kafka_rest_client.js

```diff
@@ -72,7 +74,8 @@ function KafkaRestClient(options, callback) {
     self.discoverTopics(function discoverTopicsCallback(err, isDone) {
         if (err || !isDone) {
             self.enable = false;
-            callback(new Error('Failed to discover topics mapping'));
+            self.scheduleConnect(1);
+            callback(); // Will not return an error since there are retries to rest proxy
```

Comment:
Some users complain about kafka-rest-client reports error when failing to connect to rest proxy as nodesol doesn't. Since we have retries now, users don't need to know if this connection succeeds or not. If it fails, they will receive an error about disconnection when they produce messages. We did the same change in go client as go users also complain about this.